File: services/src/main/java/com/senacor/code/fullstack/chat/RequestLoggingFilter.java
package com.senacor.code.fullstack.chat;

import org.springframework.stereotype.Component;
import org.springframework.web.filter.CommonsRequestLoggingFilter;

import javax.servlet.http.HttpServletRequest;

@Component
public class RequestLoggingFilter extends CommonsRequestLoggingFilter {

    public RequestLoggingFilter() {
        setIncludeQueryString(true);
        setIncludePayload(true);
        setMaxPayloadLength(10000);
        setIncludeHeaders(false);
        setBeforeMessagePrefix(" ");
        setBeforeMessageSuffix("");
    }

    @Override
    protected void beforeRequest(HttpServletRequest request, String message) {
        super.beforeRequest(request, request.getMethod() +  message);
    }

    @Override
    protected void afterRequest(HttpServletRequest request, String message) {
        // don't log after request
    }
}

File: services/src/main/kotlin/com/senacor/code/fullstack/chat/RequestLoggingFilter.kt
package com.senacor.code.fullstack.chat

import org.springframework.stereotype.Component
import org.springframework.web.filter.CommonsRequestLoggingFilter
import javax.servlet.http.HttpServletRequest

@Component
class RequestLoggingFilter : CommonsRequestLoggingFilter() {
    init {
        isIncludeQueryString = true
        isIncludePayload = true
        maxPayloadLength = 10000
        isIncludeHeaders = false
        setBeforeMessagePrefix(" ")
        setBeforeMessageSuffix("")
    }

    override fun beforeRequest(request: HttpServletRequest?, message: String) {
        super.beforeRequest(request, request!!.method + message)
    }

    override fun afterRequest(request: HttpServletRequest?, message: String) {
        // don't log after request
    }
}